polyfill/test/PlainDate/prototype/getFields/order-of-operations.js

```diff
@@ -41,7 +38,7 @@ Object.defineProperties(ObservedDate.prototype, {
 const instance = new ObservedDate(2000, 5, 2);
 
 const result = instance.getFields();
-assert.sameValue(result.year, 2000, "year result");
-assert.sameValue(result.month, 5, "month result");
-assert.sameValue(result.day, 2, "day result");
 assert.compareArray(actual, expected, "order of operations");
+assert.sameValue(result.year.valueOf(), 2000, "year result");
```

Comment:
Should we remove the trap on valueOf now that we're not checking for it? Or is it important to check that we _don't_ call valueOf?

Reply:
Good question, I'm not sure. As is, the current test does check that we don't call valueOf to do any coercion. I was having trouble understanding the logic around why the ISO calendar shouldn't be converting these to integers when they're custom getters for fields on a temporal object with the ISO calendar.

Looking now I actually think something might be broken (I might have broken something here). I'll dig into it.

Reply:
(Result of discussion: nothing is broken, this is correct; it's important to check that we don't call valueOf, in this case)